docs/visualize/vega.asciidoc

```diff
@@ -3,71 +3,2114 @@
+than to develop a {kib} plugin when in need of more features.
+
+[[vega-lite-tutorial]]
+=== Tutorial: First visualization in Vega-Lite
 
-You create Vega visualizations by using the text editor, which is
-preconfigured with the options you need.
+When you first open the Vega editor in {kib}, you will see a pre-populated
+line chart which shows the total number of documents across all your indices
+within the time range.
 
 [role="screenshot"]
 image::images/vega_lite_default.png[]
 
-[float]
-[[vega-schema]]
-==== Change the Vega version
+This tutorial will show you how to build a stacked area chart, and how to debug
+the most common issues when writing a visualization in Vega:
 
-The default visualization uses Vega-Lite version 2. To use Vega version 4, edit
-the `schema`.
+[role="screenshot"]
+image::images/vega_lite_tutorial_1.png[]
 
-Go to `$schema`, enter `https://vega.github.io/schema/vega/v5.json`, then click
-*Update*.
+The text editor contains a Vega-Lite spec written in https://hjson.github.io/[HJSON],
+which is similar to JSON but optimized for human editing. HJSON supports:
 
-[float]
-[[vega-type]]
-==== Change the visualization type
+* Comments using // or /* syntax
+* Object keys without quotes
+* String values without quotes
+* Optional commas
+* Double or single quotes
 
-The default visualization is a line chart. To change the visualization type,
-change the `mark` value. The supported visualization types are listed in the
-text editor.
+Multiline strings are supported in HJSON, but will not work as expected in Vega.
 
-Go to `mark`, change the value to a different visualization type, then click
-*Update*.
+The default Vega-Lite spec is querying `_all` indices, but most visualizations are powered by a single index. Change the line:
+
+```yaml
+index: _all
+```
+
+to:
+
+```yaml
+index: kibana_sample_data_ecommerce
+```
+
+and click "Update". The result is probably not what you expect, because it shows
+a flat line with 0 results. This shows the first thing that you need to understand
+when writing a Vega spec in {kib}: small changes can lead to unexpected results.
+
+In this case, you've only changed the index, so the problem must be that the results
+of your new query don't match what you expected. To debug this, you can use one of
+these techniques:
+
+* Use intuition to guess at the problem
+* Open the browser dev tools and look at the Network tab to inspect the {es} request
+* Open the browser dev tools and type `VEGA_DEBUG.vegalite_spec.data` to inspect
+    the response from {es}
+* Build a comparable query using the {kib} Dev Tools and see if the results match
+* Copy the spec from `JSON.stringify(VEGA_DEBUG.vegalite_spec)` and paste into the
+    online Vega-Lite editor
+* Use `VEGA_DEBUG.view.data('source_0')` or `VEGA_DEBUG.view.data('data_0')` to
+    understand the transformations that are applied after fetching data
+
+This complex debugging process will come up frequently while working on Vega in {kib}.
```

Comment:
I don't think we should write "this complex debugging process" - no reason to scare people ahead of time ;)